src/pages/EarningsInvoices.vue

```diff
@@ -0,0 +1,341 @@
+const SORT_PAYMENT = 'Payment Status'
+const SORT_EXCEPTION_STATUS = 'Exception Status'
+
+const FILTER_ALL = 'All'
+const FILTER_UNPAID_LATE = 'Unpaid & Late'
+const FILTER_PAID = 'Paid'
+const FILTER_EXCEPTIONS = 'Exceptions'
+
+export default {
+  name: 'EarningsInvoices',
+  components: {
+    PrimaryLayout,
+    ShortUpArrowIcon,
+    RightChevronIcon
+  },
+  data () {
+    return {
+      invoices: mockInvoiceData,
+      headers: [
+        SORT_HAPP,
+        SORT_PUBLISHER,
+        SORT_CREATED,
+        SORT_DUE,
+        SORT_INVOICE,
+        SORT_AMOUNT,
+        SORT_PAYMENT,
+        SORT_EXCEPTION_STATUS
+      ],
+      filter: FILTER_ALL,
+      sort: SORT_CREATED,
+      sortDesc: true,
+      pageSize: 10,
+      page: 0
+    }
+  },
+  created () {
+    this.filterOptions = [FILTER_ALL, FILTER_UNPAID_LATE, FILTER_PAID, FILTER_EXCEPTIONS]
+    this.pageSizeOptions = [5, 10, 20, 30, 50]
+
+    const queryFilter = this.$route.query.filter
+    if (queryFilter === 'unpaid') {
+      this.filter = FILTER_UNPAID_LATE
+    } else if (queryFilter === 'exceptions') {
+      this.filter = FILTER_EXCEPTIONS
+    }
+  },
+  computed: {
+    breadcrumbs () {
+      return [{
+        label: 'Earnings',
+        path: '/earnings'
+      }, {
+        label: 'Invoices'
+      }]
+    },
+    paginationLegend () {
+      const first = (this.page * this.pageSize) + 1
+      const last = first + this.pagedInvoices.length - 1
+      return `${first}-${last} of ${this.invoiceCount} items`
+    },
+    hasPrevPage () {
+      return this.page > 0
+    },
+    hasNextPage () {
+      return ((this.page + 1) * this.pageSize <= this.invoiceCount)
+    },
+    filteredSortedInvoices () {
+      let filtered
+      switch (this.filter) {
+        case FILTER_ALL:
+            filtered = this.invoices
+          break;
+        case FILTER_UNPAID_LATE:
+            filtered = this.invoices.filter(invoice => invoice.payment_status === PSTATUS_UNPAID || invoice.payment_status === PSTATUS_LATE)
+          break;
+        case FILTER_PAID:
+            filtered = this.invoices.filter(invoice => invoice.payment_status === PSTATUS_PAID)
+          break;
+        case FILTER_EXCEPTIONS:
+            filtered = this.invoices.filter(invoice => invoice.exception_status === ESTATUS_EXCEPTION)
+          break;
+        default:
+            filtered = this.invoices
+          break;
+      }
+
+      const sortKey = ({
+        [SORT_HAPP]: 'happ',
+        [SORT_PUBLISHER]: 'publisher',
+        [SORT_CREATED]: 'date_created',
+        [SORT_DUE]: 'date_due',
+        [SORT_INVOICE]: 'id',
+        [SORT_AMOUNT]: 'amount',
+        [SORT_PAYMENT]: 'payment_status',
+        [SORT_EXCEPTION_STATUS]: 'exception_status'
+      })[this.sort]
+
+      return filtered.sort((a, b) =>
+        this.sortDesc
+          ? a[sortKey] > b[sortKey]
+            ? -1
+            : 1
+          : a[sortKey] < b[sortKey]
+            ? -1
+            : 1)
```

Comment:
Shouldn't this return 0 if they're equal? Will having this return 1 for equal cause unwanted rearranging of the list?

Reply:
Yup, updated.